### PR TITLE
PP-670 fixes the illegal state exception and optimistic lock exception bug while cancelling AUTH_SUCCESS charges

### PR DESCRIPTION
**Pull request by rahul-vinayak**
- Added EXPIRE_CANCEL_PENDING as one of the legal states for cancelling
  a charge.
- Expanded the transaction, to cover up for optimistic lock exception
  being thrown earlier while doing charge.update calls on a detached
  objects when expiring a charge with AUTHORISATION_SUCCESS state.
- Update and merge charge calls now return the merged charge collection
  as opposed to voids, so that versions get updated in the managed charge
  entities
- updated the tests to cover this bug fix

### CODE REVIEW

File: src/test/java/uk/gov/pay/connector/it/resources/ChargeCancelFrontendResourceITest.java [hunk 1]
```diff
@@ -43,7 +43,6 @@
             CAPTURED,
             CAPTURE_SUBMITTED,
             CAPTURE_ERROR,
-            EXPIRE_CANCEL_PENDING,
             EXPIRE_CANCEL_FAILED,
```
Comment (mrlumbu) [inline src/test/java/uk/gov/pay/connector/it/resources/ChargeCancelFrontendResourceITest.java]: Should EXPIRE_CANCEL_PENDING not be added to cancellable statuses?

File: src/test/java/uk/gov/pay/connector/it/resources/ChargeCancelResourceITest.java [hunk 2]
```diff
@@ -69,7 +69,6 @@ public void respondWith400_whenNotCancellableState() {
                 CAPTURED,
                 CAPTURE_SUBMITTED,
                 CAPTURE_ERROR,
-                EXPIRE_CANCEL_PENDING,
```
Comment (mrlumbu) [inline src/test/java/uk/gov/pay/connector/it/resources/ChargeCancelResourceITest.java]: As before, should EXPIRE_CANCEL_PENDING not be added to cancellable statuses?

File: src/test/java/uk/gov/pay/connector/it/resources/ChargeCancelFrontendResourceITest.java (hunk 1 above)
Comment (rahul-vinayak) [inline src/test/java/uk/gov/pay/connector/it/resources/ChargeCancelFrontendResourceITest.java]: Probably yes if there was an ITest for that, and thats the reason I don't like separate statuses being maintained in the code and test, should be driven off the same List coming from the actual code. Will try and refactor the code a bit here

**Comment by mrlumbu** [comment]
👍

**Comment by rahul-vinayak** [comment]
rebuild

**Pull request merged**